nebula-logger/main/logger-engine/classes/LogEntryEventBuilder.cls

```diff
@@ -433,18 +440,35 @@ global with sharing class LogEntryEventBuilder {
      * @return         The same instance of `LogEntryEventBuilder`, useful for chaining methods
      */
     global LogEntryEventBuilder setRecord(List<SObject> records) {
-        if (this.shouldSave == false || records == null) {
+        if (this.shouldSave == false) {
             return this;
         }
 
-        Schema.SObjectType sobjectType = records.getSObjectType() != null ? records.getSObjectType() : records.get(0).getSObjectType();
-        String sobjectTypeName = sobjectType?.getDescribe().getName();
-
         this.logEntryEvent.RecordCollectionType__c = 'List';
-        this.logEntryEvent.RecordJson__c = getJson(records);
-        this.logEntryEvent.RecordSObjectClassification__c = getSObjectClassification(sobjectType);
-        this.logEntryEvent.RecordSObjectType__c = sobjectTypeName;
-        this.logEntryEvent.RecordSObjectTypeNamespace__c = getSObjectTypeNamespace(sobjectType);
+
+        // First, try to use the list's SObject Type
+        // If that fails, then try to use the first record's SObject Type
+        // If THAT fails, then log as is a generic SObject
```

Comment:
seems like this comment is missing a word or two - I love this method though, haha

Reply:
Whoops, yeah, I think I meant to come back & add more details here after I had finished the implementation. I'll update this comment with some more details.